### PR TITLE
[bazel] Update Bazel README with Bzlmod & WORKSPACE examples

### PR DESCRIPTION
**Pull request by ssarad**
Add an example of how to pull swift-syntax with bzlmod and WORKSPACE

### CODE REVIEW

**Comment by ssarad** [comment]
@keith Can I have you take a look as well please?

**Comment by ssarad** [comment]
@swift-ci please test

File: README.md [hunk 1]
```diff
@@ -36,8 +36,29 @@ Start contributing to SwiftSyntax see [this guide](CONTRIBUTING.md) for more inf
 ## Bazel
 
 SwiftSyntax provides an experimental [Bazel](https://bazel.build) build configuration, maintained by Keith Smiley. 
-To use it you can pull the source archive from the relevant release tag
-into your `WORKSPACE` and depend on the libraries you need from the
+To use it, you can pull the source archive from the relevant release tag
+into your `MODULE.bazel` file (preferred and recommended) with `bazel_dep`. Bzlmod support was added starting release of `509.0.0` and above. All available versions can be found in the [Bazel Central Registry](https://registry.bazel.build/)
```
[the remark this answers is not in the document]
Comment (ssarad) [reply README.md]: done!

File: README.md [hunk 2]
```diff
@@ -36,8 +36,29 @@ Start contributing to SwiftSyntax see [this guide](CONTRIBUTING.md) for more inf
...
+into your `MODULE.bazel` file (preferred and recommended) with `bazel_dep`. Bzlmod support was added starting release of `509.0.0` and above. All available versions can be found in the [Bazel Central Registry](https://registry.bazel.build/modules/swift-syntax)
+
+```python3
+bazel_dep(
+    name = "swift-syntax",
+    version = "600.0.1",
+)
+```
+
+You can also pull source archive with `WORKSPACE` prior to `509.0.0` but note that it is preferred to use `MODULE.bazel` with latest version and also `WORKSPACE` is being deprecated as of Bazel 8 and unsupported on Bazel 9. To use `WORKSPACE` and swift-syntax, you can use `http_archive` as such
```
Comment (dduan) [inline README.md:49]: this implies `509.0.0`+ can't be used with WORKSPACE with isn't true.

File: README.md [hunk 3]
```diff
@@ -36,8 +36,29 @@ Start contributing to SwiftSyntax see [this guide](CONTRIBUTING.md) for more inf
...
+
+```python3
+http_archive(
+    name = "SwiftSyntax",
+    sha256 = "f070fd44db9b33f430fd5b5d2700f1e2001c0028711859600e80cc975074fab0",
+    strip_prefix = "swift-syntax-509.1.0",
+    url = "https://github.com/apple/swift-syntax/archive/refs/tags/509.1.0.tar.gz",
+)
+
```
Comment (dduan) [inline README.md:58]: unnecessary newline

File: README.md (hunk 2 above)
Comment (ssarad) [reply README.md]: yea thats true, i updated the comment w/o the version number so that isnt implied

Comment (dduan) [reply README.md]: I also think we don't need to mention the WORKSPACE support removal. For any reason folks need to use WORKSPACE,  they will stay on Bazel 8 until those reasons are gone. Chances of those people learning about WORKSPACE removal in Bazel 9 from this doc are very low IMO.

**Comment by dduan** [comment]
@swift-ci please test

File: README.md (hunk 2 above)
Comment (ssarad) [reply README.md]: I think my thought process was to leave a note for future users to also encourage to bump to bzlmod but i see your point. We can update the docs again later if needed.  

removed the deprecated note 👍 thanks for the reviews!

**Review by dduan — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -36,8 +36,28 @@ Start contributing to SwiftSyntax see [this guide](CONTRIBUTING.md) for more inf
...
-To use it you can pull the source archive from the relevant release tag
-into your `WORKSPACE` and depend on the libraries you need from the
+To use it, you can pull the source archive from the relevant release tag
+into your `MODULE.bazel` file (preferred and recommended) with `bazel_dep`. Bzlmod support was added starting release of `509.0.0` and above. All available versions can be found in the [Bazel Central Registry](https://registry.bazel.build/modules/swift-syntax)
+
+```python3
+bazel_dep(
+    name = "swift-syntax",
+    version = "600.0.1",
+)
```
Comment (keith) [inline README.md:46]:
```suggestion
bazel_dep(name = "swift-syntax", version = "600.0.1")
```

seems to me like the community is generally preferring using this as a single line

**Review by keith — APPROVED** [review]
lgtm, i doubt we'll keep these versions up to date so it's definitely useful to have the links out to the newer versions

**Comment by ssarad** [comment]
Thanks for all the reviews! @keith @dduan , not sure why the CI tests aren't starting but anything left to merge this?

**Comment by keith** [comment]
@swift-ci please test

**Comment by ssarad** [comment]
@swift-ci please test

**Comment by keith** [comment]
@swift-ci please test windows platform

**Pull request merged**